docs/data/material/customization/overriding-component-structure/overriding-component-structure.md

```diff
@@ -0,0 +1,108 @@
+The demo below shows how to replace the Button's `<button>` tag with a `<a>` to create a link button:
+
+{{"demo": "OverridingRootSlot.js"}}
+
+:::info
+The props `href`, `target`, and `rel` are specific to `<a>` tag. Be sure to use the appropriate attributes when you provide a custom `component` prop.
+:::
+
+## Interior slots
+
+Complex components are composed of one or more interior slots in addition to the root.
+These slots are often (but not necessarily) nested within the root.
+
+For example, the [Autocomplete](/material-ui/react-autocomplete/) is composed of a root `<div>` that houses several interior slots named for the elements they represent: input, startDecorator, endDecorator, clearIndicator, popupIndicator and so on.
+
+### The slots prop
+
+Use the `slots` prop to replace a component's interior slots.
+The example below shows how to replace the popper slot in the [Autocomplete](/material-ui/react-autocomplete/) component to remove the popup functionality:
+
+{{"demo": "OverridingInternalSlot.js"}}
+
+### The slotProps prop
+
+The `slotProps` prop is an object that contains the props for all slots within a component.
+You can use it to define additional custom props to pass to a component's interior slots.
+
+For example, the code snippet below shows how to add a custom `data-testid` to the popper slot of the [Autocomplete](/material-ui/react-autocomplete/) component:
+
+```jsx
+<Autocomplete slotProps={{ popper: { 'data-testid': 'my-popper' } }} />
+```
+
+All additional props placed on the primary component are also propagated into the root slot (just as if they were placed in `slotProps.root`).
+These two examples are equivalent:
+
+```jsx
+<Autocomplete id="badge1">
+```
+
+```jsx
+<Autocomplete slotProps={{ root: { id: 'badge1' } }}>
+```
+
+:::warning
+If both `slotProps.root` and additional props have the same keys but different values, the `slotProps.root` props will take precedence.
+This does not apply to classes or the `style` prop—they will be merged instead.
+:::
+
+## Best practices
+
+Use `component` or `slotProps.{slot}.component` prop to override the element by preserving the styles of the slot.
+
+Use `slots` prop to replace the slot's styles and functionality with your custom component.
+
+Overriding with `component` lets you apply the attributes of that element directly to the root.
+For instance, if you override the Button's root with an `<li>` tag, you can add the `<li>` attribute `value` directly to the component.
+If you did the same with `slots.root`, you would need to place this attribute on the `slotProps.root` object in order to avoid a TypeScript error.
+
+Be mindful of your rendered DOM structure when overriding the slots of more complex components.
+You can easily break the rules of semantic and accessible HTML if you deviate too far from the default structure—for instance, by unintentionally nesting block-level elements inside of inline elements.
+
+## History of the pattern
```

Comment:
I've added suggestions to improve this section line by line, but I think most of it can be cut because it's not really necessary for documentation purposes. I would change this header to `## Context`, move this section to the top of the doc, after the intro but before `## The mental model`, and cut the list of benefits (this would make more sense in a blog post on the topic).

Reply:
Sounds good to me. Updated.